### PR TITLE
feat: add standalone bundle

### PR DESCRIPTION
**Pull request by magicmatatjahu**
**Description**

Changes proposed in this pull request:

- Add `@asyncapi/react-component/browser/standalone` bundle
- Add documentation how to use bundle in Angular and Vue projects
- Update docs for NextJS with section about SSR and SSG
- Add e2e tests of standalone bundle using cypress

**Related issue(s)**
See also https://github.com/asyncapi/asyncapi-react/issues/308
Closes https://github.com/asyncapi/asyncapi-react/issues/304

### CODE REVIEW

**Review by jonaslagoni — COMMENTED** [review]
I really miss some tests which ensure that our documentation actually is up to date and works. Do you see any way we can do this? Blackbox testing?

**Comment by magicmatatjahu** [comment]
@jonaslagoni Only tests and tests 😄 Don't you trust me? The new functions are only a wrapper for React API, but ok I will try to test it :)

**Comment by jonaslagoni** [comment]
> @jonaslagoni Only tests and tests 😄 Don't you trust me? The new functions are only a wrapper for React API, but ok I will try to test it :)

😈

Because you bundle stuff and suggest a way to use the library at some point something is gonna be added that alters the output in a way so it won't be possible to use it the same way as you suggest it should be integrated.

**Comment by magicmatatjahu** [comment]
@jonaslagoni Ok, I added (with huge delay 😅) e2e tests using cypress. Currently, I don't want to add e2e testing for Angular and Vue as separate tests, because each framework has different way of e2e testing, Angular for example has also another e2e testing framework. 

The current tests test it as if other frameworks were tested, that is: give component's props (like schema) as the first argument to the `render` function of bundle and as the second some DOM node, where component will be rendered :)

**Review by jonaslagoni — COMMENTED** [review]
Alright 👍 Just had 2 comments suggesting to add some jsdoc to public functions.

File: library/src/standalone-codebase.ts [hunk 1]
```diff
@@ -0,0 +1,46 @@
...
+} from 'react-dom';
+
+function querySelector(selector: string): Element | DocumentFragment | null {
+  if (typeof document !== 'undefined') {
+    return document.querySelector(selector);
+  }
+  return null;
+}
+
+export function createRender<P>(component: any) {
```
[the remark this answers is not in the document]
Comment (magicmatatjahu) [reply library/src/standalone-codebase.ts:19]: Added :) Return type is inferred by TS :)

File: library/src/standalone-codebase.ts [hunk 2]
```diff
@@ -0,0 +1,46 @@
...
+    container = container || querySelector('asyncapi');
+    if (container === null) {
+      return;
+    }
+
+    renderComponent(React.createElement(component, props), container, callback);
+  };
+}
+
+export function createHydrate<P>(component: any) {
```
[the remark this answers is not in the document]
Comment (magicmatatjahu) [reply library/src/standalone-codebase.ts:39]: Added :) Return type is inferred by TS :)

**Review by jonaslagoni — APPROVED** [review]
👍

**Comment by magicmatatjahu** [comment]
@jonaslagoni Thanks a lot! :)

**Pull request merged**